pysr/regressor_sequence.py

```diff
@@ -0,0 +1,257 @@
+        index : int | list[int]
+            If you want to compute the output of an expression using a
+            particular row of `self.equations_`, you may specify the index here.
+            For multiple output equations, you must pass a list of indices
+            in the same order.
+        extra_predictions : int
+            If you want to predict more than one step into the future, specify
+            how many extra predictions you want. For example, if `extra_predictions=2`,
+            the model will predict the next two time points after the last time point
+            in `X`.
+
+        Returns
+        -------
+        x_predicted : ndarray of shape (n_samples, n_features)
+            Values predicted by substituting `X` into the fitted sequence symbolic
+            regression model.
+
+        Raises
+        ------
+        ValueError
+            Raises if the `best_equation` cannot be evaluated.
+        """
+        X = self._validate_data(X)
+        _check_assertions(X, recursive_history_length=self.recursive_history_length)
+        historical_X = np.lib.stride_tricks.sliding_window_view(
+            X.flatten(), self.recursive_history_length * np.prod(X.shape[1])
+        )[:: X.shape[1], :]
+        pred = self._regressor.predict(X=historical_X, index=index)
+        if extra_predictions > 0:
+            output = pred
+            previous_points = historical_X[-1]
+            # Without this, the model will re-predict the last data point
+            pred_once = self._regressor.predict(X=[previous_points], index=index)
+            previous_points = previous_points[X.shape[1] :]
+            previous_points = np.append(previous_points, pred_once)
+            previous_points = previous_points.flatten()
+            for _ in range(extra_predictions):
+                pred_once = self._regressor.predict(X=[previous_points], index=index)
+                previous_points = previous_points[X.shape[1] :]
+                previous_points = np.append(previous_points, pred_once)
+                previous_points = previous_points.flatten()
+                output = np.append(output, pred_once)
+            return output.reshape(-1, X.shape[1])
+        return pred
+
+    def __getattr__(self, name):
+        # return self._regressor.__getattr__(name)
+        try:
+            return getattr(self._regressor, name)
+        except AttributeError:
+            raise AttributeError(
+                f"'{self.__class__.__name__}' object has no attribute '{name}'"
+            )
+
+    def __delattr__(self, name):
+        if name == "_regressor":
+            print("no delete regressor")
+        else:
+            delattr(self._regressor, name)
+
+    def __setattr__(self, name, value) -> None:
+        if name == "_regressor":
+            object.__setattr__(self, name, value)
+        else:
+            setattr(self._regressor, name, value)
+            print(self._regressor.__dict__[name])
```

Comment:
I think we should just define each method individually and forward arguments, like

```python
    def sympy(self, index=None):
        return self._regressor.sympy(index=index)
```

This is much safer, as it forces us to think about which functions are compatible and which are not. For example I think `.pytorch` and `.jax` are not compatible (and it's not worth it to get them working), so we should just leave those unimplemented.

Reply:
how do I forward properties like `equations_`?

Reply:
```
@property
def equations_
    return self._regressor.equations_
```